rust/src/router/consumer.rs

```diff
@@ -395,6 +589,55 @@ enum Notification {
     Trace(ConsumerTraceEventData),
 }
 
+impl Notification {
+    pub(crate) fn from_fbs(
+        notification: notification::NotificationRef<'_>,
+    ) -> Result<Self, NotificationParseError> {
+        match notification.event().unwrap() {
+            notification::Event::ConsumerProducerClose => Ok(Notification::ProducerClose),
+            notification::Event::ConsumerProducerPause => Ok(Notification::ProducerPause),
+            notification::Event::ConsumerProducerResume => Ok(Notification::ProducerResume),
+            notification::Event::ConsumerScore => {
+                let Ok(Some(notification::BodyRef::ConsumerScoreNotification(body))) =
+                    notification.body()
+                else {
+                    panic!("Wrong message from worker: {notification:?}");
+                };
+
+                let score_fbs = consumer::ConsumerScore::try_from(body.score().unwrap()).unwrap();
+                let score = ConsumerScore::from_fbs(score_fbs);
+
+                Ok(Notification::Score(score))
+            }
+            notification::Event::ConsumerLayersChange => {
+                let Ok(Some(notification::BodyRef::ConsumerLayersChangeNotification(body))) =
+                    notification.body()
+                else {
+                    panic!("Wrong message from worker: {notification:?}");
+                };
+
+                let layers_fbs =
+                    consumer::ConsumerLayers::try_from(body.layers().unwrap()).unwrap();
+                let layers = ConsumerLayers::from_fbs(layers_fbs);
+
+                Ok(Notification::LayersChange(Some(layers)))
+            }
+            notification::Event::ConsumerTrace => {
+                let Ok(Some(notification::BodyRef::ConsumerTraceNotification(body))) =
+                    notification.body()
+                else {
+                    panic!("Wrong message from worker: {notification:?}");
+                };
+
+                let trace_notification_fbs = consumer::TraceNotification::try_from(body).unwrap();
+                let trace_notification = ConsumerTraceEventData::from_fbs(trace_notification_fbs);
+
+                Ok(Notification::Trace(trace_notification))
+            }
+            _ => Err(NotificationParseError::InvalidEvent),
+        }
+    }
+}
```

Comment:
New line between structs/impls, please (here and in other places):
```suggestion
}

```